services/Postgres.go

```diff
@@ -43,8 +44,120 @@ func (postgres PostgresService) GetCycleTimeSeconds() int {
 	return postgres.Config.CycleTimeSeconds
 }
 
-func (postgres PostgresService) ComputeScalingProposal(s.ScaledObject) (s.ResourceScalingProposal, error) {
-	return s.ResourceScalingProposal{}, nil // TODO: implement
+func (postgres PostgresService) ComputeScalingProposal(object s.ScaledObject) (s.ResourceScalingProposal, error) {
+	var cluster *s.Cluster
+	switch objectType := object.(type) {
+	case *s.Cluster:
+		cluster = objectType
+	default:
+		return s.ResourceScalingProposal{}, fmt.Errorf("unsupported scaled object type: %s", object.GetType())
+	}
+	if !cluster.Ready {
+		return s.ResourceScalingProposal{}, fmt.Errorf("cluster %s (%s) is not ready", cluster.ClusterName, cluster.ClusterId)
+	}
+	return postgres.applyRules(*cluster), nil
+}
+
+func (postgres PostgresService) applyRules(cluster s.Cluster) s.ResourceScalingProposal {
```

Comment:
The name is a bit misleading. It should be named calculateScalingProposal or somethin like that

Reply:
computeScalingProposalInternal or computeTargetResources

Reply:
Done